rust/worker/src/segment/record_segment.rs

```diff
@@ -499,12 +493,18 @@ impl<'a> SegmentWriter<'a> for RecordSegmentWriter {
             .unwrap()
             .commit::<u32, &DataRecord>()
             .await;
-        let flusher_max_offset_id = self
-            .max_offset_id
-            .take()
-            .unwrap()
-            .commit::<&str, u32>()
-            .await;
+        let max_offset_id = self.max_offset_id.take().unwrap();
+        let max_known_offset_id = self.max_new_offset_id.load(atomic::Ordering::SeqCst);
+        if max_known_offset_id > 0 {
```

Comment:
Are we using the 0 state as a way to indicate no change? May be useful to comment that here 
https://github.com/chroma-core/chroma/pull/3192/files#diff-f589c7d82e5323e79007dc1cf1e115249559a8f8ffd7ad274d0c590d8dcbdcb9R298

Reply:
Yes. I've added a few comments to document this.